src/Fetch.js

```diff
@@ -345,14 +345,12 @@ function fetchXHR(fetch, onsuccess, onerror, onprogress, onreadystatechange) {
     {{{ makeSetValue('fetch', C_STRUCTS.emscripten_fetch_t.readyState, 'xhr.readyState', 'i16') }}}
     {{{ makeSetValue('fetch', C_STRUCTS.emscripten_fetch_t.status, 'xhr.status', 'i16') }}}
     if (xhr.statusText) stringToUTF8(xhr.statusText, fetch + {{{ C_STRUCTS.emscripten_fetch_t.statusText }}}, 64);
-    if (fetchAttrSynchronous)
+    if (fetchAttrSynchronous) {
       // The response url pointer malloc()ed here has the same lifetime as the emscripten_fetch_t structure itself has, and is
       // freed when emscripten_fetch_close() is called.
       var ruPtr = stringToNewUTF8(xhr.responseURL);
-      if (ruPtr) {
-        _free(fetch + {{{ C_STRUCTS.emscripten_fetch_t.responseUrl }}});
-        {{{ makeSetValue('fetch', C_STRUCTS.emscripten_fetch_t.responseUrl, 'ruPtr', '*') }}}
-      }
+      _free(fetch + {{{ C_STRUCTS.emscripten_fetch_t.responseUrl }}});
```

Comment:
I don't think you need this free call either.. since its only assigned once isn't it?

Reply:
You're right. I did not do it on previous commit because I was still checking the standard. But now I have normative reference that in whatwg/xhr section 3.5.6 step 11.9 that this will only happen exactly once unless an error occurred before reaching this step when it will not be triggered at all. I will remove the free's.